model_bakery/random_gen.py

```diff
@@ -94,67 +136,70 @@ def num_as_str(x):
 gen_decimal.required = ["max_digits", "decimal_places"]
 
 
-def gen_date():
+def gen_date() -> date:
     return now().date()
 
 
-def gen_datetime():
+def gen_datetime() -> datetime:
     return now()
 
 
-def gen_time():
+def gen_time() -> time:
     return now().time()
 
 
-def gen_string(max_length):
+@action_generator
```

Comment:
Not sure I follow exactly why we are putting this workaround under certain methods, which don't have direct attributes. But I need to take a deeper look here and will play with the code deeper.

Reply:
This annotation is on all the generators that have `.required` set. We could probably add it to the others, but it wasn't needed for type checking, so I hadn't done so yet

Reply:
I have mixed feelings about this. For now, for example, if an user wants to define a custom generator function, they'll only have to do something like:

```
def custom_gen_func():
    return "random value"

baker.generators.add('test.generic.fields.CustomField', custom_gen_func)
```

By introducing this new `@action_generator` the user will have an extra step that's to remind to decorate the function with it. Also, I'm now sure how will it work with existing test.

Reply:
Hadn't thought about the external generator case. Right, I've killed off action_generator entirely. It was a nice idea, but the external case makes it unusable. I've added ignores for the existing usages of all the properties so mypy won't complain about them (which reduces the checking a bit, but still better to ignore some things and check others than not check at all)